AppCenterAnalytics/AppCenterAnalytics/Internals/MSAnalytics+Validation.m

```diff
@@ -59,4 +66,70 @@ - (nullable NSString *)validateEventName:(NSString *)eventName forLogType:(NSStr
+  MSEventProperties *validCopy = [MSEventProperties new];
+  for (NSString *propertyKey in eventProperties.properties) {
+    if ([validCopy.properties count] == kMSMaxPropertiesPerLog) {
+      MSLogWarning([MSAnalytics logTag], @"Typed properties cannot contain more than %i items. Skipping other properties.", kMSMaxPropertiesPerLog);
+      break;
+    }
+    MSTypedProperty *property = eventProperties.properties[propertyKey];
+    MSTypedProperty *validProperty = [self validateAppCenterTypedProperty:property];
+    if (validProperty) {
+      validCopy.properties[validProperty.name] = validProperty;
+    }
+  }
+  return validCopy;
+}
+
+- (MSTypedProperty *)validateAppCenterTypedProperty:(MSTypedProperty *)typedProperty {
+  MSTypedProperty *validProperty;
+  if ([typedProperty isKindOfClass:[MSStringTypedProperty class]]) {
+    MSStringTypedProperty *originalStringProperty = (MSStringTypedProperty *)typedProperty;
+    MSStringTypedProperty *validStringProperty = [MSStringTypedProperty new];
+    validStringProperty.value = [self validateAppCenterStringTypedPropertyValue:originalStringProperty.value];
+    validProperty = validStringProperty;
+  } else if ([typedProperty isKindOfClass:[MSBooleanTypedProperty class]]) {
+    validProperty = [MSBooleanTypedProperty new];
+    ((MSBooleanTypedProperty *)validProperty).value =  ((MSBooleanTypedProperty *)typedProperty).value;
+  } else if ([typedProperty isKindOfClass:[MSLongTypedProperty class]]) {
+    validProperty = [MSLongTypedProperty new];
+    ((MSLongTypedProperty *)validProperty).value =  ((MSLongTypedProperty *)typedProperty).value;
+  } else if ([typedProperty isKindOfClass:[MSDoubleTypedProperty class]]) {
+    validProperty = [MSDoubleTypedProperty new];
+    ((MSDoubleTypedProperty *)validProperty).value =  ((MSDoubleTypedProperty *)typedProperty).value;
+  } else if ([typedProperty isKindOfClass:[MSDateTimeTypedProperty class]]) {
+    validProperty = [MSDateTimeTypedProperty new];
+    ((MSDateTimeTypedProperty *)validProperty).value =  ((MSDateTimeTypedProperty *)typedProperty).value;
+  }
+
+
+  validProperty.name = [self validateAppCenterPropertyName:typedProperty.name];
+  return validProperty;
+}
+
+- (NSString *)validateAppCenterPropertyName:(NSString *)propertyKey {
+  if ([propertyKey length] > kMSMaxPropertyKeyLength) {
+    MSLogWarning([MSAnalytics logTag], @"Typed property '%@': key length cannot exceed %i characters. Property value will be truncated.", propertyKey,
+                 kMSMaxPropertyKeyLength);
+    return [propertyKey substringToIndex:(kMSMaxPropertyKeyLength - 1)];
+  }
+  return propertyKey;
+}
+
+- (NSString *)validateAppCenterStringTypedPropertyValue:(NSString *)value {
+  if ([value length] > kMSMaxPropertyValueLength) {
+    MSLogWarning([MSAnalytics logTag], @"Typed property value length cannot exceed %i characters. Property value will be truncated.",
```

Comment:
Let's match this up with the property name validation and print out the value in the warning message.